src/node/desktop/src/main/detect-r.ts

```diff
@@ -55,34 +58,56 @@ function executeCommand(command: string): Expected<string> {
   }
 }
 
-export async function promptUserForR(): Promise<Expected<string | null>> {
-  // nothing to do if RSTUDIO_WHICH_R is set
-  const rstudioWhichR = getenv('RSTUDIO_WHICH_R');
-  if (rstudioWhichR) {
-    return ok(rstudioWhichR);
-  }
+export async function promptUserForR(platform = process.platform): Promise<Expected<string | null>> {
+  if (platform === 'win32') {
+    const desktop = await import('../native/desktop.node');
 
-  // discover available R installations
-  const rInstalls = findRInstallationsWin32();
-  if (rInstalls.length === 0) {
-    return err();
-  }
+    const storeConfig = new Store({ defaults: kDesktopOptionDefaults });
+    const rstudioPathKey = 'rstudioPath';
 
-  // ask the user what version of R they'd like to use
-  const dialog = new ChooseRModalWindow(rInstalls);
-  const [path, error] = await dialog.showModal();
-  if (error) {
-    return err(error);
-  }
+    const isCtrlKeyDown = desktop.isCtrlKeyDown() as boolean;
```

Comment:
Is the `as boolean` here necessary? Since we already declare the return value in the `.d.ts` file.

Reply:
Fixed! This was mostly required from my old code.